README.md

```diff
@@ -983,6 +984,105 @@ This section describes code style for [ECMAScript 2015 Language Specification](h
 
 [⬆ back to TOC](#table-of-contents)
 
+### Arrow functions
+
+* [Arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+  should be used where an anonymous function is expected (when you need function and you don't want bind it to an
+  identifier):
+
+  > Explanation:
+  > * `this` keyword in arrow function binds to the upper scope's `this`. It prevents usage invalid `this` or usage less
+  >    efficient `Function.prototype.bind` function.
+  > * The notation is shorter than function expression.
+
+  ```js
+  [1, 2, 3].map((x) => {
+      // ...
+  });
+
+  // Use `function` here, because pass `someObj` as `this` argument.
+  [1, 2, 3].map(function (x) {
+      // ...
+  }, someObj);
+  ```
+
+* If you bind function with some name, use `function declaration`:
```

Comment:
It would be nice to add an explanation for this case. Why exactly is it bad?
```
const isGoodEnough = (x) => x > this._x;
someArray.filter(isGoodEnough);
anotherArray.filter(isGoodEnough);
```
?